instrumentation/opentelemetry-instrumentation-aws-lambda/tests/test_aws_lambda_instrumentation_manual.py

```diff
@@ -414,3 +414,12 @@ def test_uninstrument(self):
         mock_execute_lambda(MOCK_LAMBDA_API_GATEWAY_HTTP_API_EVENT)
         spans = self.memory_exporter.get_finished_spans()
         self.assertEqual(len(spans), 0)
+
+    def test_no_op_tracer_provider(self):
+        tracer_provider = NoOpTracerProvider()
+        AwsLambdaInstrumentor().instrument(tracer_provider=tracer_provider)
+
+        mock_execute_lambda(MOCK_LAMBDA_API_GATEWAY_HTTP_API_EVENT)
+
+        spans = self.memory_exporter.get_finished_spans()
```

Comment:
please verify "spans" to avoid "object of type 'NoneType' has no len" err

Reply:
BTW get_finished_spans returns a tuple, so you cant get None

https://github.com/open-telemetry/opentelemetry-python/blob/c0e8f4056db774e4eeaf3f30016a19b2ba8b0408/opentelemetry-sdk/src/opentelemetry/sdk/trace/export/in_memory_span_exporter.py#L43

Reply:
Since python can be dynamic typing I feel better with 'not None' validation but I agree it's a negotiable :)